backend/ipfs/filemanager.go

```diff
@@ -0,0 +1,301 @@
+	"github.com/ipfs/boxo/exchange"
+	"github.com/ipfs/boxo/files"
+	"github.com/ipfs/boxo/ipld/merkledag"
+	unixfile "github.com/ipfs/boxo/ipld/unixfs/file"
+	"github.com/ipfs/boxo/ipld/unixfs/importer/balanced"
+	"github.com/ipfs/boxo/ipld/unixfs/importer/helpers"
+	"github.com/ipfs/boxo/provider"
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+	multihash "github.com/multiformats/go-multihash"
+	"go.uber.org/zap"
+)
+
+const (
+	// IPFSRootRoute is the root route of ipfs.
+	IPFSRootRoute = "/ipfs"
+	// UploadRoute is the route to upload a file.
+	UploadRoute = "/file-upload"
+	routeVar    = "cid"
+	// GetRoute is the route to get a file.
+	GetRoute = "/{" + routeVar + "}"
+	// MaxFileMB is the maximum file size (in MB) to be uploaded.
+	MaxFileMB = 64
+	// SearchTimeout is the maximum time we are searching for a file.
+	SearchTimeout = 2 * time.Minute
+)
+
+// AddParams contains all of the configurable parameters needed to specify the
+// importing process of a file.
+type AddParams struct {
+	Layout    string
+	Chunker   string
+	RawLeaves bool
+	Hidden    bool
+	Shard     bool
+	NoCopy    bool
+	HashFun   string
+}
+
+// HTTPHandler is an interface to pass to the router only the http handlers and
+// not all the FileManager type.
+type HTTPHandler interface {
+	GetFile(http.ResponseWriter, *http.Request)
+	UploadFile(http.ResponseWriter, *http.Request)
+}
+
+// FileManager is the main object to handle ipfs files.
+type FileManager struct {
+	ctx        context.Context
+	started    bool
+	log        *zap.Logger
+	exch       exchange.Interface
+	bstore     blockstore.Blockstore
+	bservice   blockservice.BlockService
+	DAGService ipld.DAGService // become a DAG service
+	provider   provider.System
+}
+
+// NewManager creates a new fileManager instance.
+func NewManager(ctx context.Context, log *zap.Logger) *FileManager {
+	return &FileManager{
+		log:     log,
+		ctx:     ctx,
+		started: false,
+	}
+}
+
+// Start starts new manager.
+func (fm *FileManager) Start(blockstore blockstore.Blockstore, bitswap *Bitswap, provider provider.System) error {
+	fm.bstore = blockstore
+	fm.exch = bitswap
+	fm.provider = provider
+	if err := fm.setupBlockService(); err != nil {
+		return err
+	}
+
+	if err := fm.setupDAGService(); err != nil {
+		fm.bservice.Close()
+		return err
+	}
+
+	go fm.autoclose()
+	fm.started = true
+	return nil
+}
+
+func (fm *FileManager) autoclose() {
+	<-fm.ctx.Done()
+	fm.bservice.Close()
+	fm.started = false
+}
+
+func (fm *FileManager) setupBlockService() error {
+	fm.bservice = blockservice.New(fm.bstore, fm.exch)
+	return nil
+}
+
+func (fm *FileManager) setupDAGService() error {
+	fm.DAGService = merkledag.NewDAGService(fm.bservice)
+	return nil
+}
+
+// GetFile retrieves a file from ipfs.
+func (fm *FileManager) GetFile(w http.ResponseWriter, r *http.Request) {
+	encoder := json.NewEncoder(w)
+	if !fm.started {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		_ = encoder.Encode("IPFS node not started")
```

Comment:
@juligasa  You don't need to use JSON encoder here if you don't write any JSON back. You could have used `fmt.Fprintf(w, ...)`.